### PR TITLE
Add Jenkins example using Diffblue Cover Optimize [TG-16885]

### PR DESCRIPTION
**Pull request by jeannielynnmoulton**

### CODE REVIEW

File: jenkins-github-maven/optimize/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                    # Get and activate dcover, export required variable to point to dcover script location
+                    getDcover "$RELEASE_URL"
+                    activateDcover "$LICENSE_KEY"
+                    export DIFFBLUE_COMMAND="$(realpath $(getDcoverScriptLocation))"
+
+                    # Generate the patch file, export required variable to point to patch file
+                    export DIFFBLUE_PATCH="$(realpath $(patchFile))"
+                    git diff origin/jenkins/base-branch-optimize > $DIFFBLUE_PATCH
+
+                    # Call mvn test, only tests affected by the changes in the patch file are run now
```
Comment (peterschrammel) [inline jenkins-github-maven/optimize/Jenkinsfile]: You can also call `mvn verify`

File: jenkins-github-maven/optimize/scripts/common.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+# Copyright 2021-2022 Diffblue Limited. All Rights Reserved.
+# Unpublished proprietary source code.
+# Use is governed by https://docs.diffblue.com/licenses/eula
+
+####### dcover jars #######
+
+# This should retrieve and unzip all of the scripts and jars required for dcover and echo the
+# the location of the dcover script. In this example, they are unzipped into the directory dcover.
```
Comment (peterschrammel) [inline jenkins-github-maven/optimize/scripts/common.sh:6]: double `the`

File: jenkins-github-maven/optimize/scripts/common.sh [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+# This should also set up the license if required, but that is not done here as it depends your
+# your particular agreement with Diffblue.
+# In you modify this, be sure to modify getDcoverScriptLocation to be correct.
```
Comment (peterschrammel) [inline jenkins-github-maven/optimize/scripts/common.sh]: In -> If

File: jenkins-github-maven/optimize/scripts/common.sh [hunk 4]
```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+# Copyright 2021-2022 Diffblue Limited. All Rights Reserved.
+# Unpublished proprietary source code.
```
Comment (peterschrammel) [inline jenkins-github-maven/optimize/scripts/common.sh]: We should release these scripts under a more permissive license, e.g. Apache 2.0. We don't really need to restrict their use to licensed customers.

File: jenkins-github-maven/optimize/scripts/common.sh [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  then
+    echoDiffblue "The last command failed."
+    exit 1
+  fi
+}
+
+echoDiffblue() {
+  MESSAGE="$1"
+  echo "[DIFFBLUE CI PIPELINE] $MESSAGE"
+}
```
Comment (peterschrammel) [inline jenkins-github-maven/optimize/scripts/common.sh]: newline

File: jenkins-github-maven/optimize/scripts/common.sh (hunk 2 above)
Comment (jeannielynnmoulton) [reply jenkins-github-maven/optimize/scripts/common.sh:6]: This is like the puzzle:

I love Paris in the
the Springtime

File: jenkins-github-maven/optimize/scripts/common.sh (hunk 4 above)
Comment (jeannielynnmoulton) [reply jenkins-github-maven/optimize/scripts/common.sh]: I've deleted this for now. I'll update all of the example with the same thing in another PR. Do we have wording for this already somewhere?

File: jenkins-github-maven/optimize/scripts/common.sh (hunk 2 above)
Comment (peterschrammel) [reply jenkins-github-maven/optimize/scripts/common.sh:6]: echo cho ho o?

File: jenkins-github-maven/optimize/scripts/common.sh (hunk 4 above)
Comment (peterschrammel) [reply jenkins-github-maven/optimize/scripts/common.sh]: See bottom of this page: https://www.apache.org/licenses/LICENSE-2.0

**Review by peterschrammel — APPROVED** [review]

**Review by jennysykes — APPROVED** [review]

**Pull request merged**